### PR TITLE
chore(multi-env): core unittest supports multi-env

### PR DESCRIPTION
**Pull request by a1exwang**
The whole plan is to make the unit test support both multi-env enabled/disabled cases. And then run the unit test twice: with TEAMSFX_INSIDER_PREVIEW and without.

- fix the unit test so the it can pass with multi-env enabled in fx-core
  - use deepEqual to compare JSON because the key order of JSON.stringify is not [in any particular order](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
  - throw the error instead of `assert.fail` so we can know the actual stack and error message when the test fails 
- fix a projectMigrator bug
- fix loop import in core

### CODE REVIEW

**Review by LongOddCode — APPROVED** [review]

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -176,3 +176,44 @@ jobs:
...
+
+      - name: Setup project
+        run: |
+          npm run setup
+
+      - name: Unit Test
+        id: unit-test
+        shell: bash
+        run: |
+          export TEAMSFX_PRIVATE_PREVIEW=true
```
Comment (dooriya) [inline .github/workflows/unit-test.yml:210]: Should it be `TEAMSFX_INSIDER_PREVIEW`?

Comment (a1exwang) [reply .github/workflows/unit-test.yml]: Fixed.

File: .github/workflows/unit-test.yml [hunk 2]
```diff
@@ -176,3 +176,44 @@ jobs:
...
+      - name: Setup project
+        run: |
+          npm run setup
+
+      - name: Unit Test
+        id: unit-test
+        shell: bash
+        run: |
+          export TEAMSFX_INSIDER_PREVIEW=true
+          # replace with this eventually: xvfb-run -a npx lerna run test:unit
```
[the remark this answers is not in the document]
Comment (a1exwang) [reply .github/workflows/unit-test.yml:211]: export only affects current bash process, and it does not affect global envs. What do you mean by "re-run behavior"?

[the remark this answers is not in the document]
Comment (dooriya) [reply .github/workflows/unit-test.yml:211]: You've explained already. Thanks!

**Review by qinezh — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**